File: LeetcodeSolutions/Hashing/ContainsDuplicate.java
import java.util.HashSet;
public class ContainsDuplicate {
    public static void main(String[] args) {
        int[] nums = new int[] {1,2,3,1};
        System.out.println(containsDuplicate(nums));
    }

    public static boolean containsDuplicate(int[] nums) {
        HashSet<Integer> set = new HashSet<>();

        for (int num: nums) {
            if (set.contains(num)) return true;
            set.add(num);
        }

        return false;
    }
}

File: LeetcodeSolutions/Hashing/ContainsDuplicate.py
def containsDuplicate(nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        hashset = set()
        for i in nums:
            if i in hashset:
                return True
            hashset.add(i)
        return False

nums = [1, 2, 3, 1]
print(containsDuplicate(nums))